conformance/lang/expressions/error_constructor.balt

```diff
@@ -0,0 +1,526 @@
+        io:println(err.detail()); // @output {"no":100}
+    }
+}
+
+Test-Case: output
+Description: Test 5 for error constructor when the contextually expected type is a union.
+Fail-Issue: ballerina-platform/ballerinalang#29345
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, intersection-type, union-type, error:detail,
+        method-call-expr, string
+
+type E1 error<record {int code;}>;
+
+type E2 error<record {int no;}>;
+
+type E E1 & E2;
+
+function init() {
+    E|E1|E2 err1 = error("msg", code = 100, no = 100);
+    io:println(err1.detail()); // @output {"code":100,"no":100}
+}
+
+Test-Case: error
+Description: Test 6 for error constructor when the contextually expected type is a union.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, distinct-type, union-type, string, byte
+
+type E1 distinct error<record {int code;}>;
+
+type E2 distinct error<record {byte code;}>;
+
+function errorFunction() {
+    E1|E2 err1 = error("msg", code = 100); // @error cannot infer type of the error from '(E1|E2)'
+
+    byte x = 100;
+    E1|E2 err2 = error("msg", code = x); // @error cannot infer type of the error from '(E1|E2)'
+}
+
+Test-Case: output
+Description: Test 7 for error constructor when the contextually expected type is a union.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, distinct-type, union-type, string, is-expr
+
+type E1 distinct error<record {int code;}>;
+
+type E2 distinct error<record {byte code;}>;
+
+function init() {
+    int x = 100;
+    E1|E2 err = error("msg", code = x);
+
+    io:println(err is E1); // @output true
+    io:println(err is E2); // @output false
+}
+
+Test-Case: output
+Description: Test type-ids of the constructed error value are those induced by applicable contextually expected type.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, distinct-type, intersection-type, string,
+        is-expr
+
+type E1 distinct error<record {int code;}>;
+
+type E2 distinct error<record {int no;}>;
+
+type E3 error<record {int code; int no;}>;
+
+type E E1 & E2;
+
+function init() {
+    E1 err1 = error("message", code = 100);
+    error err2 = err1;
+    io:println(err2 is E1); // @output true
+
+    E err3 = error("message", code = 100, no = 200);
+    E3 err4 = err3;
+    io:println(err4 is E); // @output true
+}
+
+Test-Case: output
+Description: Test nested error constructor.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, string, method-call-expr, if-else-stmt,
+        optional-type, is-expr, error:message, error:cause, error:detail
+
+type E error<record {|string message?; boolean fatal?;|}>;
+
+function init() {
+    error err1 = error("message 1", error E("message 2", message = "ERROR!", fatal = true));
+    io:println(err1.message()); // @output message 1
+    io:println(err1.detail()); // @output {}
+
+    error? cause1 = err1.cause();
+    if cause1 is error {
```

Comment:
Shall we do `io:println(cause1 is error);` prior to `if-else`?
Let's add in other places too.